src/engine/source/builder/builders/opBuilderSHAfrom.cpp

```diff
@@ -0,0 +1,128 @@
+#include "opBuilderSHAfrom.hpp"
+
+#include <optional>
+#include <string>
+
+#include <fmt/format.h>
+#include <json/json.hpp>
+#include <openssl/evp.h>
+#include <openssl/sha.h>
+
+#include "baseTypes.hpp"
+#include <baseHelper.hpp>
+
+namespace
+{
+
+// Sha1 digest len (20) * 2 (hex chars per byte)
+constexpr int OS_SHA1_HEXDIGEST_SIZE = (SHA_DIGEST_LENGTH * 2);
+constexpr int OS_SHA1_ARRAY_SIZE_LEN = OS_SHA1_HEXDIGEST_SIZE + 1;
+
+std::optional<std::string> wdbi_strings_hash(const std::vector<std::string>& input)
+{
+    char* parameter = NULL;
+    unsigned char digest[EVP_MAX_MD_SIZE];
+    unsigned int digest_size;
+
+    EVP_MD_CTX* ctx = EVP_MD_CTX_create();
+    if (!ctx)
+    {
+        // Failed during hash context creation
+        return std::nullopt;
+    }
+
+    if (1 != EVP_DigestInit(ctx, EVP_sha1()))
+    {
+        // Failed during hash context initialization
+        EVP_MD_CTX_destroy(ctx);
+        return std::nullopt;
+    }
+
+    for (const auto& word : input)
+    {
+        if (1 != EVP_DigestUpdate(ctx, word.c_str(), word.length()))
+        {
+            // Failed during hash context update
+            return std::nullopt;
+        }
+    }
+
+    EVP_DigestFinal_ex(ctx, digest, &digest_size);
+    EVP_MD_CTX_destroy(ctx);
```

Comment:
From [SSL Doc:](https://www.openssl.org/docs/man1.0.2/man3/EVP_DigestInit.html
)

> The functions EVP_DigestInit(), EVP_DigestFinal() and EVP_MD_CTX_copy() are obsolete but are retained to maintain compatibility with existing code. 
> New applications should use EVP_DigestInit_ex(), EVP_DigestFinal_ex() and EVP_MD_CTX_copy_ex() because  they can efficiently reuse a digest context instead of initializing and cleaning it up on each call and allow non default implementations of digests to be specified.

Reply:
great catch!